Services/Init/classes/class.ilStartUpGUI.php

```diff
@@ -1039,165 +1039,151 @@ protected function purgePlaceholders($page_editor_html)
+        $form->addCommandButton('migrateAccount', $this->lng->txt('save'));
+        $form->addCommandButton('showLogin', $this->lng->txt('cancel'));
 
-        $rad = new ilRadioGroupInputGUI($lng->txt('auth_account_migration_name'), 'account_migration');
+        $rad = new ilRadioGroupInputGUI($this->lng->txt('auth_account_migration_name'), 'account_migration');
         $rad->setValue(1);
 
         $keep = new ilRadioOption(
-            $lng->txt('auth_account_migration_keep'),
+            $this->lng->txt('auth_account_migration_keep'),
             static::ACCOUNT_MIGRATION_MIGRATE,
-            $lng->txt('auth_info_migrate')
+            $this->lng->txt('auth_info_migrate')
         );
-        $user = new ilTextInputGUI($lng->txt('login'), 'mig_username');
+        $user = new ilTextInputGUI($this->lng->txt('login'), 'mig_username');
         $user->setRequired(true);
-        $user->setValue(ilUtil::prepareFormOutput($_POST['mig_username']));
+        $user->setValue(ilUtil::prepareFormOutput(
+            (string) ($this->httpRequest->getParsedBody()['mig_username'] ?? '')
+        ));
         $user->setSize(32);
         $user->setMaxLength(128);
         $keep->addSubItem($user);
 
-        $pass = new ilPasswordInputGUI($lng->txt('password'), 'mig_password');
+        $pass = new ilPasswordInputGUI($this->lng->txt('password'), 'mig_password');
         $pass->setRetype(false);
         $pass->setRequired(true);
-        $pass->setValue(ilUtil::prepareFormOutput($_POST['mig_password']));
+        $pass->setValue(ilUtil::prepareFormOutput(
+            (string) ($this->httpRequest->getParsedBody()['mig_password'] ?? '')
+        ));
         $pass->setSize(12);
         $pass->setMaxLength(128);
         $keep->addSubItem($pass);
         $rad->addOption($keep);
 
         $new = new ilRadioOption(
-            $lng->txt('auth_account_migration_new'),
+            $this->lng->txt('auth_account_migration_new'),
             static::ACCOUNT_MIGRATION_NEW,
-            $lng->txt('auth_info_add')
+            $this->lng->txt('auth_info_add')
         );
         $rad->addOption($new);
 
         $form->addItem($rad);
 
         $tpl->setVariable('MIG_FORM', $form->getHTML());
 
-        if (strlen($a_message)) {
-            ilUtil::sendFailure($a_message);
+        if (strlen($message)) {
+            ilUtil::sendFailure($message);
         }
 
-        $tpl->printToStdout('DEFAULT');
+        self::printToGlobalTemplate($tpl);
     }
 
     /**
      * Migrate Account
      * @return bool
      */
-    protected function migrateAccount()
+    protected function migrateAccount() : bool
     {
-        if (!isset($_POST['account_migration'])) {
+        if (!isset($this->httpRequest->getParsedBody()['account_migration'])) {
             $this->showAccountMigration(
-                $GLOBALS['DIC']->language()->txt('err_choose_migration_type')
+                $this->lng->txt('err_wrong_login')
```

Comment:
I will change this to txt->('select_one') afterwards.

Reply:
Thx